### ISSUE TITLE
Include FDR q value in command line output

### ISSUE THREAD

**Issue by thinkle12** [mpc-bioinformatics/pia#105]
Hi all, its me again...

I was wondering if it is possible to include "FDR q values" in the `-proteinExport` output from command line. I suppose I can calculate FDR myself by just simply counting decoys based on a score threshold. However, I just wanted to point this out. The output from command line I see is as follows:
```
"accessions","score","#peptides","#PSMs","#spectra"
"USP9X_HUMAN|Q93008","374.75325142812","155","450","438"
"USP9Y_HUMAN|O00507","141.30843342618374","66","138","138"
"Q6P468_HUMAN|Q6P468","139.16527025922363","59","170","164"
"HSP7C_HUMAN|P11142,V9HW22_HUMAN|V9HW22","110.67590820388949","47","111","111"
"GRP78_HUMAN|P11021,V9HWB4_HUMAN|V9HWB4","99.65650591097625","41","90","90"
"A0A0G2JIW1_HUMAN|A0A0G2JIW1,A8K5I0_HUMAN|A8K5I0,HS71A_HUMAN|P0DMV8,HS71B_HUMAN|P0DMV9","95.12141332736277","41","103","103"
"B7Z4V2_HUMAN|B7Z4V2,GRP75_HUMAN|P38646,V9HW84_HUMAN|V9HW84","87.67906153604937","40","71","71"
"A0A087WZG9_HUMAN|A0A087WZG9,B4DSP0_HUMAN|B4DSP0,PEG10_HUMAN|Q86TG7-2","78.44328812851143","31","358","187"
"A0A087WUL4_HUMAN|A0A087WUL4,A0A087WX23_HUMAN|A0A087WX23,A0A087WXK2_HUMAN|A0A087WXK2,PEG10_HUMAN|Q86TG7","77.50461134847883","31","358","187"
```

Within Knime this information is output as follows:
```
Proteins | Score | Coverages | nrPeptides | nrPSM | nrSpectra | ClusterID | Description | Decoy | FDR q value
[USP9X_HUMAN\|Q93008] | 433.7986176 | [?] | 152 | 363 | 426 | 11547 | [Probable ubiquitin carboxyl-terminal hydrolase FAF-X OS=Homo sapiens   GN=USP9X PE=1 SV=3] | FALSE | 0
[USP9Y_HUMAN\|O00507] | 175.9273045 | [?] | 66 | 128 | 136 | 11547 | [Probable ubiquitin carboxyl-terminal hydrolase FAF-Y OS=Homo sapiens   GN=USP9Y PE=2 SV=2] | FALSE | 0
[Q6P468_HUMAN\|Q6P468] | 160.3044466 | [?] | 56 | 137 | 159 | 11547 | [USP9X protein (Fragment) OS=Homo sapiens GN=USP9X PE=2 SV=1] | FALSE | 0
[GRP78_HUMAN\|P11021, V9HWB4_HUMAN\|V9HWB4] | 122.2464199 | [?, ?] | 42 | 80 | 86 | 68 | [78 kDa glucose-regulated protein OS=Homo sapiens GN=HSPA5 PE=1 SV=2,   Epididymis secretory sperm binding protein Li 89n OS=Homo sapiens   GN=HEL-S-89n PE=2 SV=1] | FALSE | 0
[HSP7C_HUMAN\|P11142, V9HW22_HUMAN\|V9HW22] | 121.8969185 | [?, ?] | 44 | 97 | 105 | 68 | [Heat shock cognate 71 kDa protein OS=Homo sapiens GN=HSPA8 PE=1 SV=1,   Epididymis luminal protein 33 OS=Homo sapiens GN=HEL-S-72p PE=2 SV=1] | FALSE | 0
[A0A0G2JIW1_HUMAN\|A0A0G2JIW1,   A8K5I0_HUMAN\|A8K5I0, HS71A_HUMAN\|P0DMV8, HS71B_HUMAN\|P0DMV9] | 110.6020159 | [?, ?, ?, ?] | 39 | 86 | 98 | 68 | [Heat shock 70 kDa protein 1B OS=Homo sapiens GN=HSPA1B PE=1 SV=1,   Epididymis secretory protein Li 103 OS=Homo sapiens GN=HSPA1A PE=2 SV=1, Heat   shock 70 kDa protein 1A OS=Homo sapiens GN=HSPA1A PE=1 SV=1, Heat shock 70   kDa protein 1B OS=Homo sapiens GN=HSPA1B PE=1 SV=1] | FALSE | 0
[B7Z4V2_HUMAN\|B7Z4V2, GRP75_HUMAN\|P38646,   V9HW84_HUMAN\|V9HW84] | 109.0415013 | [?, ?, ?] | 38 | 62 | 67 | 68 | [cDNA FLJ51907, highly similar to Stress-70 protein, mitochondrial   OS=Homo sapiens PE=2 SV=1, Stress-70 protein, mitochondrial OS=Homo sapiens   GN=HSPA9 PE=1 SV=2, Epididymis secretory sperm binding protein Li 124m   OS=Homo sapiens GN=HEL-S-124m PE=2 SV=1] | FALSE | 0
[B4DNT8_HUMAN\|B4DNT8] | 100.6255528 | [?] | 36 | 77 | 88 | 68 | [cDNA FLJ54370, highly similar to Heat shock 70 kDa protein 1 OS=Homo   sapiens PE=2 SV=1] | FALSE | 0

```

Where the ending column is the FDR q value. 

Also, How would it be possible to get coverage results? I believe I have seen a workflow that uses OpenMS PeptideIndexer to do this?

Thanks,

Trent

**Comment by julianu** [comment]
Hej Trent, right, that should be included into the command line output as well.

To get coverage results, the complete protein sequence must be given in the search results. Some formats contain these, some don't. But right, using OpenMS's PeptideIndexer is one easy way to solve this issue.

**Comment by thinkle12** [comment]
Hi @julianu, Could you also potentially add ClusterID as well to the command line output? Currently without that we do not see proper subset grouping. Meaning in the PIa protein output, I see many cases where we have 1 protein ie (USP9X) and an isoform of it (USP9X-1). The entirety of (USP9X) in terms of peptides from the search are also in (USP9X-1). However, (USP9X-1) has 1 unique peptide. After PIA we wish to group these together because essentially we want to treat USP9X as being a subset of USP9X-1, and only to report 1 total protein here as opposed to two. I have seen that while using Knime these two will be clustered together with the same ClusterID. Therefore, if you could also add ClusterID into the command line output that would also be great.

**Comment by julianu** [comment]
Hej @thinkle12 ,
I will make the CSV export essentially the same as the KNIME tables, or at least containing all information. Stay tuned, will inform you on any progress at this site.

**Comment by julianu** [comment]
This should be fixed now by #119
